wgpu-hal/src/dx12/device.rs

```diff
@@ -620,6 +620,16 @@ impl crate::Device<super::Api> for super::Device {
                 .anisotropy_clamp
                 .map_or(0, |_| d3d12::D3D12_FILTER_ANISOTROPIC);
 
+        let border_color = if desc
+            .address_modes
+            .iter()
+            .any(|am| am == &wgt::AddressMode::ClampToZero)
+        {
+            [0.0; 4]
+        } else {
+            conv::map_border_color(desc.border_color)
```

Comment:
oh snap! Actually we have a problem here.
What happens if a sampler has both `ClampToZero` and `ClampToBorder` modes? We can't set the border color to zero in this case. It's a conceptual problem, and I'm not seeing an easy solution

Reply:
I think the only solution is to declare that this is not allowed and panic

Reply:
> I think a solution would require this new behavior to be a part of enum SamplerBorderColor enum (e.g. AutoBlack)

Actually, I think this would work, but it's not too different from just writing a workaround in the Metal backend where it falls back to using ClampToZero if ClampToBorder is not supported. What do you think of this approach vs the one you descibed?

Reply:
Please clarify your suggestion. Are you saying, let's expose this feature on 100% metal, and just document that the border color may be ignored on platforms that don't support it? I'm concerned that this would make it problematic for people who need a specific color, and they'd lose an ability to discover if it's supported. Frankly, I don't know right off the bat who would need that.

Reply:
You're right that not having the ability to check whether it's supported is not ideal. So then we'd have to go back to my initial suggestion, which is to declare that using `ClampToZero` and `ClampToBorder` at the same time is a validation error.

Reply:
I still think it's better to have this as an enum value for `SamplerBorderColor`, which would encode this validation rule by construction.

Reply:
Alright, I'll do that.